### PR TITLE
Support HSE and USB clock

### PR DESCRIPTION
**Pull request by agalakhov**

### CODE REVIEW

File: src/rcc.rs [hunk 1]
```diff
@@ -221,50 +235,67 @@ impl CFGR {
...
 
+        // set prescalers and clock source
+        rcc.cfgr.modify(|_, w| unsafe {
+            w.ppre2()
+                .bits(ppre2_bits)
+                .ppre1()
+                .bits(ppre1_bits)
+                .hpre()
+                .bits(hpre_bits)
+                .usbpre()
```
Comment (dfrankland) [inline src/rcc.rs]: Seems like the `stm32f301` and the `stm32f334` don't have a USB prescalers. If you'd like, I can fix this up with some conditional configuration.

**Comment by dfrankland** [comment]
Thank you so much for contributing this! :smile:

File: src/rcc.rs (hunk 1 above)
Comment (agalakhov) [reply src/rcc.rs]: Thank you, I was not aware of this.

Comment (agalakhov) [reply src/rcc.rs]: I added one more commit for feature-gating this.

**Comment by dfrankland** [comment]
Awesome work, thanks again @agalakhov

**Pull request merged**